File: Chapter-21/FieldGoals.py
###########################################################################
### Chapter 21 - Football Decision Making 101                           ###
# Mathletics: How Gamblers, Managers, and Sports Enthusiasts              #
# Use Mathematics in Baseball, Basketball, and Football                   #
###########################################################################

import pandas as pd
import numpy as np
import statsmodels.formula.api as smf
import seaborn as sns
import matplotlib.pyplot as plt

dataset = pd.read_csv("fieldgoals.csv")

## run the logistic regression model
f = 'Good~Distance'
logitfit = smf.logit(formula = str(f), data = dataset).fit()

print(logitfit.summary2())

## visualize the results 

sns.set_style(style='white') 
ax = plt.subplot(111)
ax = sns.regplot(x="Distance", y="Good", data=dataset, logistic = True)
plt.ylabel("Success Probability")
plt.savefig("FG.png")

File: Chapter-21/FieldGoals.R
###########################################################################
### Chapter 21 - Football Decision Making 101                           ###
# Mathletics: How Gamblers, Managers, and Sports Enthusiasts              #
# Use Mathematics in Baseball, Basketball, and Football                   #
###########################################################################

library(readr)
library(stats)
library(ggplot2)

dataset <- read_csv("fieldgoals.csv")

## run the logistic regression model
f <- as.formula("Good ~ Distance")
logitfit <- glm(f, data = dataset, family = binomial())

print(summary(logitfit))

## visualize the results
ggplot(dataset, aes(x = Distance, y = Good)) +
  geom_point() +
  geom_smooth(method = "glm", method.args = list(family = "binomial"), se = FALSE) +
  ylab("Success Probability")
ggsave("FG.png")

